src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_skeram.cpp

```diff
@@ -158,9 +160,11 @@ class boss_skeram : public CreatureScript
                             events.ScheduleEvent(EVENT_ARCANE_EXPLOSION, urand(8000, 18000));
                             break;
                         case EVENT_FULLFILMENT:
-                            /// @todo For some weird reason boss does not cast this
-                            // Spell actually works, tested in duel
-                            DoCast(SelectTarget(SELECT_TARGET_RANDOM, 0, 0.0f, true), SPELL_TRUE_FULFILLMENT, true);
+                            if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 1))
+                            {
+                                DoCast(target, SPELL_TRUE_FULFILLMENT);
+                                DoCast(target, SPELL_GENERIC_DISMOUNT, true);
```

Comment:
as you can see, SPELL_GENERIC_DISMOUNT is cast after the 500ms casttime, move it to spellscript too